src/desktop/space/element.rs

```diff
@@ -34,53 +35,95 @@ impl From<RenderZindex> for Option<u8> {
 }
 
 /// Trait for a space element
-pub trait SpaceElement<R, E>
-where
-    R: Renderer + ImportAll,
-    E: RenderElement<R>,
-{
-    /// Gets the location of this element on the specified space
-    fn location(&self, space_id: usize) -> Point<i32, Logical>;
-    /// Gets the geometry of this element on the specified space
-    fn geometry(&self, space_id: usize) -> Rectangle<i32, Logical>;
+pub trait SpaceElement: IsAlive {
+    /// Returns the geometry of this element.
+    ///
+    /// Defaults to be equal to it's bounding box.
+    fn geometry(&self) -> Rectangle<i32, Logical> {
+        self.bbox()
+    }
+    /// Returns the bounding box of this element
+    fn bbox(&self) -> Rectangle<i32, Logical>;
+    /// Returns whenever a given point inside this element will be able to receive input
+    fn input_region(&self, point: &Point<f64, Logical>) -> bool;
```

Comment:
The name `input_region` sounds like it returns some kind of input region, maybe we can find a better name

Reply:
Possibly something like `in_input_region`?

Reply:
went for `is_in_input_region`